lambdas/evaluate-gpg45-scores/src/main/java/uk/gov/di/ipv/core/evaluategpg45scores/EvaluateGpg45ScoresHandler.java

```diff
@@ -129,13 +132,33 @@ public Map<String, Object> handleRequest(JourneyRequest event, Context context)
                 return JOURNEY_VCS_NOT_CORRELATED.toObjectMap();
             }
 
-            return checkForMatchingGpg45Profile(
+            List<SignedJWT> gpg45Credentials = new ArrayList<>();
+
+            for (SignedJWT credential : credentials) {
+                if (!VcHelper.isOperationalProfileVc(credential)) {
+                    gpg45Credentials.add(credential);
+                }
+            }
+
+            JourneyResponse checkForMatchingGpg45Profile =
+                    checkForMatchingGpg45Profile(
                             vcStoreItems,
                             ipvSessionItem,
                             clientOAuthSessionItem,
-                            credentials,
-                            ipAddress)
-                    .toObjectMap();
+                            gpg45Credentials,
+                            ipAddress);
+
+            if (configService.enabled(CoreFeatureFlag.INHERITED_IDENTITY)) {
+                if (checkForMatchingGpg45Profile.equals(JOURNEY_MET)) {
```

Comment:
This condition is a bit weird - I'd prefer to apply this 'directly' when matching a profile (rather than checking the journey response).

Can we make that method return the matched profile (or just true/false) and move the routing-specific logic here? - Then this check can be `if (metProfile) { ... }` or similar